CMakeLists.txt

```diff
@@ -155,9 +155,24 @@ if(DD4HEP_USE_EDM4HEP)
 #  DD4HEP_SETUP_EDM4HEP_TARGETS()
 endif()
 
-IF(DD4HEP_USE_HEPMC3)
+if(DD4HEP_USE_HEPMC3)
   find_package(HepMC3 REQUIRED)
-ENDIF()
+
+  # Optional dependencies for compressed input support
+  if(${HEPMC3_VERSION} VERSION_GREATER_EQUAL "3.02.05")
+    option(DD4HEP_HEPMC3_COMPRESSION_SUPPORT "Enable reading of compressed HepMC files" ON)
```

Comment:
I think we need to turn this off by default, or it will break existing installation recipes

Reply:
```suggestion
    option(DD4HEP_HEPMC3_COMPRESSION_SUPPORT "Enable reading of compressed HepMC files" OFF)
```